src/App.tsx

```diff
@@ -1,16 +1,86 @@
+import { useEffect, useState } from "react";
+import {
+  BrowserRouter as Router,
+  Routes,
+  Route,
+  Navigate,
+} from "react-router-dom";
 import "./App.css";
-import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
 import LoginPage from "./pages/LoginPage";
 import SignupPage from "./pages/SignupPage";
 import UserProfilePage from "./pages/UserProfilePage";
+import EditDetailsPage from "./pages/EditDetailsPage";
+import Navbar from "./components/Navbar";
+import { onAuthStateChanged } from "firebase/auth";
+import { auth } from "./firebase";
+import { useDispatch } from "react-redux";
+import { AppDispatchType, RootState } from "./redux/Store";
+import { fetchUserDetails } from "./redux/NavBarSlice";
+import ProtectedRoute from "./ProtectedRoute";
+import { Box, CircularProgress } from "@mui/material";
+import { useSelector } from "react-redux";
```

Comment:
always import all library on the top, the all our components and services @Rishab-innovative

Reply:
this is for all pages @Rishab-innovative